src/qililab/platform/platform.py

```diff
@@ -465,6 +469,66 @@ def get_parameter(self, alias: str, parameter: Parameter, channel_id: ChannelID
+        """From the runcard retrieve the parameters necessary to draw the qprogram."""
+        data_osci = {}
+        # All the aliases of the platform in a list
+        alias = [a["alias"] for a in map(asdict, self.runcard.buses)]
+        buses = [self.buses.get(alias=a) for a in alias]
+        instruments = {
+            instrument for bus in buses for instrument in bus.instruments if isinstance(instrument, (QbloxModule))
+        }
+        if all(isinstance(instrument, QbloxModule) for instrument in instruments):
+            for bus in buses:
+                for instrument, _ in zip(bus.instruments, bus.channels):
+                    if isinstance(instrument, QbloxModule):
+                        data_osci[bus.alias] = {}
+                        static_offset_i = 0
+                        static_offset_q = 0
+                        param = [
+                            Parameter.IF,
+                            Parameter.GAIN_I,
+                            Parameter.GAIN_Q,
+                            Parameter.OFFSET_I,
+                            Parameter.OFFSET_Q,
+                            Parameter.HARDWARE_MODULATION,
+                        ]
+                        for p in param:
+                            val = self.get_parameter(bus.alias, p)
+                            data_osci[bus.alias][p] = val
+
+                        data_osci[bus.alias]["instrument_name"] = instrument.name.value
+
+                        if instrument.name == InstrumentName.QBLOX_QCM:
+                            # retrieve the set offset and assign it to the bus
+                            identifier = bus.channels
+                            for awg in instrument.awg_sequencers:
+                                if awg.identifier == identifier[0]:
+                                    for idx, out in enumerate(awg.outputs):
+                                        if idx == 0:
+                                            static_offset_i = instrument.out_offsets[out]
+                                        elif idx == 1:
+                                            static_offset_q = instrument.out_offsets[out]
+                            data_osci[bus.alias]["static_offset_i"] = static_offset_i
+                            data_osci[bus.alias]["static_offset_q"] = static_offset_q
+                        else:
+                            data_osci[bus.alias]["static_offset_i"] = 0
+                            data_osci[bus.alias]["static_offset_q"] = 0
+        else:
+            raise NotImplementedError("The drawing feature is currently only supported for QBlox.")
+
+        data_oscillocope = {}
+        for bus, bus_param in data_osci.items():
+            data_oscillocope[bus] = {}
+            for param, value in bus_param.items():
+                try:
+                    data_oscillocope[bus][param.value] = value
+
+                except AttributeError:
+                    data_oscillocope[bus][param] = value
```

Comment:
Why do we copy the data from `data_osci` to a new dictionary? We could return the first dictionary directly.